### PR TITLE
Use redux in react components

### PR DESCRIPTION
**Pull request by mrkamin**
What does this PR do?
This PR is raised to submit for code review in the Microvers dashboard of Mohammad Rafi Amin.

Description of the task to be completed 👎

I will be building my "BookStore" website for more than a whole week. below are the tasks that will be done.

Project 1: Initialize the project with components (React only).
Project 2: Add reducers and actions.
**Project 3: Use Redux in React components.**
Project 4: Connect to API.
Project 5: Styling.

### CODE REVIEW

File: src/Components/BooksInputs.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        const payload = { title, author, id: uuidv4() };
+        dispatch(addBook(payload));
+        e.target.reset();
+      }}
+      >
+        <input type="text" placeholder="Book title.." id="title" required />
+        <input type="text" placeholder="Book author..." id="author" required />
+        <button type="submit">Add Book</button>
+      </form>
+    </div>
```
Comment (julie-ify) [inline src/Components/BooksInputs.js:24]: - [x] Your inputs should controlled. That means your input values should be changed using local state with the `onChange` event listener. You can visit [this article](https://blog.logrocket.com/controlled-vs-uncontrolled-components-in-react/) for reference

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @mrkamin,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Components/BooksInputs.js (hunk 1 above)
Comment (mrkamin) [reply src/Components/BooksInputs.js:24]: > * [x]  Your inputs should controlled. That means your input values should be changed using local state with the `onChange` event listener. You can visit [this article](https://blog.logrocket.com/controlled-vs-uncontrolled-components-in-react/) for reference

Hi, @julie-ify 
thanks for your valuable feedback, but I want to say that I read the project requirements Again, I couldn't find anything there that said (the inputs should be controlled). for reference below is the project requirement. 
 thanks a lot.

https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components_v2.md

Project requirements
1- Connect your store to the main component of your React app.
2 -Define a state slice which will contain your books
// Initial state:
[
  {
    "item_id": "item1",
    "title": "The Great Gatsby",
    "author": "John Smith",
    "category": "Fiction"
  },
  {
    "item_id": "item2",
    "title": "Anna Karenina",
    "author": "Leo Tolstoy",
    "category": "Fiction"
  },
  {
    "item_id": "item3",
    "title": "The Selfish Gene",
    "author": "Richard Dawkins",
    "category": "Nonfiction"
  }
]
3 -Display your books, received from the slice, in a reusable component
4 -Dispatch actions using useDispatch
5- Add a <Button> component, which includes: An event handler that adds a book to the books array (with attributes id, title and author)
7 -Add a <Button> component, which includes: An event handler that removes a book from the books array (by id)
8- Add a <Button> component, which includes: An event handler that check status (i.e add a functionality for "Check status" button click)

File: src/Components/BooksInputs.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const BooksInputs = () => {
+  const dispatch = useDispatch();
+
+  return (
+    <div>
+      <form onSubmit={(e) => {
+        e.preventDefault();
+        const title = e.target.elements.title.value;
+        const author = e.target.elements.author.value;
```
Comment (youmari) [inline src/Components/BooksInputs.js:14]: - [x] as the previous reviewer suggested you should use the local state `useState`  that will get the text `onChange` event 
using the dom element to get the value is not good practice and as you know one of the requirements is to keep you code safe and clean and not redundant so please make sure to do that

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @mrkamin , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
## Highlights 
✔️ PRO PR 
✔️ No linter errors

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/Components/BooksInputs.js (hunk 2 above)
Comment (mrkamin) [reply src/Components/BooksInputs.js:14]: @youmari,
thanks for your good and valuable feedback.
I will consider your comments and follow the instruction thanks again.

**Pull request closed without merging**